From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Enno Boland <gottox@voidlinux.org>
Date: Fri, 14 Jul 2023 23:06:40 +0200
Subject: [PATCH] Fix includes for standard compliance

- time_t is defined by <time.h>
- malloc is defined by <stdlib.h>

Fixes build against musl libc.

Co-authored-by: Duncaen <duncaen@voidlinux.org>
Co-authored-by: psykose <alice@ayaya.dev>
Origin: Apline & Void Linux
Upstream-Status: Unknown
[ismael@sourcemage.org: Normalised patch and added description]
Signed-off-by: Ismael Luceno <ismael@sourcemage.org>
---
 drawbar.c   |    1 +
 procdbase.c |    2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

--- a/procdbase.c
+++ b/procdbase.c
@@ -39,7 +39,7 @@
 #include <fcntl.h>
 #include <unistd.h>
 #include <string.h>
-#include <malloc.h>
+#include <stdlib.h>
 
 #include "atop.h"
 #include "photoproc.h"
--- a/drawbar.c
+++ b/drawbar.c
@@ -110,6 +110,7 @@
 #include <unistd.h>
 #include <regex.h>
 #include <sys/utsname.h>
+#include <time.h>
 
 #include "atop.h"
 #include "showgeneric.h"
